internal/backends/winit/renderer/femtovg.rs

```diff
@@ -433,7 +431,6 @@ impl super::WinitCompatibleCanvas for FemtoVGCanvas {
     }
 
     fn resize_event(&mut self, size: PhysicalWindowSize) {
-        self.size = size;
         self.opengl_context.ensure_resized(size)
```

Comment:
does that mean that the opengl_context might have the wrong size?

But it is really strange that we don't get the resize_event before a draw.

Maybe this is the queue of events:

1. draw (previous frame)
2. resize event
3. draw (current frame)

but when we process the draw event, the window already has been resized.

Just a theory, but a bit strange that window.inner_size() is already updated before we have time to draw, i thought it would be done in a single thread so events should be mixed like that.  unless the call to inner_size() actually do a call to the display server.

Reply:
> does that mean that the opengl_context might have the wrong size?

Yes, using the last value from the resize event results in the OpenGL context having the wrong size.

> But it is really strange that we don't get the resize_event before a draw.
> 
> Maybe this is the queue of events:
> 
> 1. draw (previous frame)
> 2. resize event
> 3. draw (current frame)
> 
> but when we process the draw event, the window already has been resized.
> 
> Just a theory, but a bit strange that window.inner_size() is already updated before we have time to draw, i thought it would be done in a single thread so events should be mixed like that. unless the call to inner_size() actually do a call to the display server.

Yes, it's strange. And it's worse on wayland, see https://github.com/slint-ui/slint/issues/1671 .